lib/chef-dk/skeletons/code_generator/recipes/cookbook.rb

```diff
@@ -24,10 +24,32 @@
                   'All Rights Reserved'
                 end
 
+cookbook_urls = case context.git
+                when 'default'
+                  <<-EOF
+# The `issues_url` points to the location where issues for this cookbook are
+# tracked.  A `View Issues` link will be displayed on this cookbook's page when
+# uploaded to a Supermarket.
+#
+# issues_url 'https://github.com/<insert_org_here>/#{context.cookbook_name}/issues'
+# The `source_url` points to the development repository for this cookbook.  A
+# `View Source` link will be displayed on this cookbook's page when uploaded to
+# a Supermarket.
+#
+# source_url 'https://github.com/<insert_org_here>/#{context.cookbook_name}'
+EOF
+                else
+                  <<-EOF
+issues_url '#{URI.join(context.git, 'issues')}' if respond_to?(:issues_url)
```

Comment:
This only makes sense with GitHub but you aren't actually checking that anywhere.

Reply:
Hi there, appreciate you giving me some feedback.  Please bear with me, what is it you object too? How I build the urls if ‘— git-url’ is not ‘default’ ?

Reply:
I would check if the provided URL is actually at `github.com` or otherwise looks like a GitHub URL. Also it would be nice to make sure the URL is well-formed though that's less of a problem and more of an easy enhancement.

Reply:
Hello again, appreciate you taking the time to help me out with this one.

I didn’t opt for parsing the url to see if was a github one, we use git enterprise internally and that doesn’t have git within fqdn. I guess by naming the argument `--git-url` we are implying the use of git and I liked being able to pass it single argument and have it generate both attributes.  

That does mean we don’t currently support the ability to allow the user to individually specify both the `issues_url` & `source_url` but do you think that should be a separate issue if that requirements comes up? I was trying to cover off the issue and my requirement but appreciate its a bigger project then that :)

I also tried to do the right thing by validating my inputs, where I was checking if the url was well-formed and feedback to the user if not.  Please let me know if you think it could be better.

https://github.com/username-is-already-taken2/chef-dk/blob/3f1cd6582eed56d3bc84eacce48868292f2001e5/lib/chef-dk/command/generator_commands/cookbook.rb#L232-L243

again appreciate your help

Best Regards

Gary

Reply:
I'd be tempted to call the option `--github-url`, since the url formats are completely specific to github (ie, they don't necessarily work for bitbucket or gitlab, etc). I'd also add options that support `issues_url` and `source_url` directly, and then make it so that passing `--github-url` just fills out those two options.
That way this is relatively easily extensible for other Git hosting providers.